### ISSUE TITLE
iOS客户端清晰度切换怎么实现？

### ISSUE THREAD

**Issue by elasticolifan** [Bilibili/ijkplayer#4522]
看了b站iOS的客户端，视频清晰度可以无缝切换 ，切换成功画面直接变清晰。优酷客户端也有类似功能。不知道这个是怎么实现的。

**Comment by tian9246** [comment]
你是想问dash 协议么？

**Comment by elasticolifan** [comment]
> 你是想问dash 协议么？

不是,我说的是手动切换清晰度.通常切换清晰度其实就是换一个播放地址从新开始播放. iOS这边ijkplayer的操作逻辑是先关闭并移除之前的播放器,然后新建一个播放器并设置为新清晰度视频的url.这样播放器有一个销毁和重建的过程,切换过程中会黑屏一会. 但我看了b站的iOS客户端,还有优酷的iOS客户端,手动切换清晰度似乎是无缝的,直接画面就变清晰了,也没有黑屏的过程.不知道这个是怎么实现的.

**Comment by drivedreams** [comment]
我有个思路 不知道可不可行：优酷的视频是m3u8的分段视频，每个小段时间很短，只要做到可以动态变更这些小段地址就可以了。如何变更呢？已经应用到播放器的地址是不能变的，变了肯定会有加载过程。那么就需要动态拦截这个地址或者地址本身就是个动态的地址。可以在本地搭建一个127.0.0.1:xxxx的简易服务器；应用给播放器的M3u8内的分段视频地址实际是通过本地服务器转发的，到了这一步就可以动态修改分段视频原地址了。

**Comment by zzzkj** [comment]
请问这个问题解决了吗？

**Comment by JerryZhou** [comment]
dash 就可以

**Comment by JerryZhou** [comment]
av_select_stream

**Comment by chicwx** [comment]
> > 你是想问dash 协议么？
> 
> 不是,我说的是手动切换清晰度.通常切换清晰度其实就是换一个播放地址从新开始播放. iOS这边ijkplayer的操作逻辑是先关闭并移除之前的播放器,然后新建一个播放器并设置为新清晰度视频的url.这样播放器有一个销毁和重建的过程,切换过程中会黑屏一会. 但我看了b站的iOS客户端,还有优酷的iOS客户端,手动切换清晰度似乎是无缝的,直接画面就变清晰了,也没有黑屏的过程.不知道这个是怎么实现的.

dash可以实现
